qutip/tests/test_floquet.py

```diff
@@ -32,24 +32,267 @@
 ###############################################################################
 
 import numpy as np
-import qutip
-
-
-def test_unitary_evolution_two_level_system():
-    delta = 1.0 * 2 * np.pi
-    eps0 = 1.0 * 2 * np.pi
-    A = 0.5 * 2 * np.pi
-    omega = np.sqrt(delta**2 + eps0**2)
-    T = 2*np.pi / omega
-    tlist = np.linspace(0, 2*T, 101)
-    psi0 = qutip.rand_ket(2)
-    H0 = -0.5*eps0*qutip.sigmaz() - 0.5*delta*qutip.sigmax()
-    H1 = 0.5 * A * qutip.sigmax()
-    args = {'w': omega}
-    H = [H0, [H1, lambda t, args: np.sin(args['w'] * t)]]
-    e_ops = [qutip.num(2)]
-
-    trial = qutip.fsesolve(H, psi0, tlist, e_ops, T, args).expect[0]
-    expected = qutip.mesolve(H, psi0, tlist, [], e_ops, args).expect[0]
-
-    np.testing.assert_allclose(trial, expected, atol=1e-4)
+from numpy.testing import assert_, run_module_suite
+from qutip import fsesolve, sigmax, sigmaz, rand_ket, num, mesolve
+from qutip import sigmap, sigmam, floquet_master_equation_rates, expect, Qobj
+from qutip import floquet_modes, floquet_modes_table, fmmesolve
+from qutip import floquet_modes_t_lookup
+
+
+class TestFloquet:
+    """
+    A test class for the QuTiP functions for Floquet formalism.
+    """
+
+    def testFloquetUnitary(self):
+        """
+        Floquet: test unitary evolution of time-dependent two-level system
+        """
+
+        delta = 1.0 * 2 * np.pi
+        eps0 = 1.0 * 2 * np.pi
+        A = 0.5 * 2 * np.pi
+        omega = np.sqrt(delta ** 2 + eps0 ** 2)
+        T = (2 * np.pi) / omega
+        tlist = np.linspace(0.0, 2 * T, 101)
+        psi0 = rand_ket(2)
+        H0 = - eps0 / 2.0 * sigmaz() - delta / 2.0 * sigmax()
+        H1 = A / 2.0 * sigmax()
+        args = {'w': omega}
+        H = [H0, [H1, lambda t, args: np.sin(args['w'] * t)]]
+        e_ops = [num(2)]
+
+        # Solve schrodinger equation with floquet solver
+        sol = fsesolve(H, psi0, tlist, e_ops, T, args)
+
+        # Compare with results from standard schrodinger equation
+        sol_ref = mesolve(H, psi0, tlist, [], e_ops, args)
+
+        assert_(max(abs(sol.expect[0] - sol_ref.expect[0])) < 1e-4)
```

Comment:
This sort of thing is better spelled as
```python
np.testing.assert_allclose(sol.expect[0], sol_ref.expect[0], atol=1e-4)
```
because then on a failure we'll see exactly what was in both arrays, and how much they differ by.